### ISSUE TITLE
Writable streams are never closed when using web workers

### ISSUE THREAD

**Issue by 0f-0b** [gildas-lormeau/zip.js#341]
As a reproduction, the following code does not log `close` to the console.

```js
const zip = new Uint8Array([
  0x50, 0x4b, 0x03, 0x04, 0x14, 0x00, 0x02, 0x00,
  0x08, 0x00, 0xb1, 0x15, 0xea, 0x54, 0x2e, 0xf9,
  0xb0, 0x5d, 0x19, 0x00, 0x00, 0x00, 0xb0, 0x04,
  0x00, 0x00, 0x09, 0x00, 0x00, 0x00, 0x68, 0x65,
  0x6c, 0x6c, 0x6f, 0x2e, 0x74, 0x78, 0x74, 0xcb,
  0x48, 0xcd, 0xc9, 0xc9, 0x57, 0x28, 0xcf, 0x2f,
  0xca, 0x49, 0xe1, 0xca, 0x18, 0x65, 0x8f, 0xb2,
  0x47, 0xd9, 0xa3, 0xec, 0x41, 0xcc, 0x06, 0x00,
  0x50, 0x4b, 0x01, 0x02, 0x1e, 0x03, 0x14, 0x00,
  0x02, 0x00, 0x08, 0x00, 0xb1, 0x15, 0xea, 0x54,
  0x2e, 0xf9, 0xb0, 0x5d, 0x19, 0x00, 0x00, 0x00,
  0xb0, 0x04, 0x00, 0x00, 0x09, 0x00, 0x00, 0x00,
  0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
  0xa4, 0x81, 0x00, 0x00, 0x00, 0x00, 0x68, 0x65,
  0x6c, 0x6c, 0x6f, 0x2e, 0x74, 0x78, 0x74, 0x50,
  0x4b, 0x05, 0x06, 0x00, 0x00, 0x00, 0x00, 0x01,
  0x00, 0x01, 0x00, 0x37, 0x00, 0x00, 0x00, 0x40,
  0x00, 0x00, 0x00, 0x00, 0x00,
]);
const reader = new ZipReader(new Uint8ArrayReader(zip));
for await (const entry of reader.getEntriesGenerator()) {
  const writable = new WritableStream({
    close() {
      console.log("close");
    },
  });
  await entry.getData(new WritableStreamWriter(writable));
}
```

**Comment by gildas-lormeau** [comment]
I agree that `WritableStreamWriter` needs more work and more tests. I don't recommend using it right now.

**Comment by gildas-lormeau** [comment]
It's fixed in the version 2.5.9.

**Comment by gildas-lormeau** [comment]
Forthe record, you can now write `await entry.getData({ writable });` instead of `await entry.getData(new WritableStreamWriter(writable));` since the version 2.5.18
